IMAGES.rst

```diff
@@ -0,0 +1,304 @@
+It is optimised for rebuild speed (AIRFLOW_CONTAINER_CI_OPTIMISED_BUILD flag set to "true").
+It installs PIP dependencies from the current branch first - so that any changes in setup.py do not trigger
+reinstalling of all dependencies. There is a second step of installation that re-installs the dependencies
+from the latest sources so that we are sure that latest dependencies are installed.
+
+The production image is a multi-segment image. The first segment "airflow-build-image" contains all the
+build essentials and related dependencies that allow to install airflow locally. By default the image is
+build from a released version of Airflow from Github, but by providing some extra arguments you can also
+build it from local sources. This is particularly useful in CI environment where we are using the image
+to run Kubernetes tests. See below for the list of arguments that should be provided to build
+production image from the local sources.
+
+Note! Breeze by default builds production image from local sources. You can change it's behaviour by
+providing ``--install-airflow-version`` parameter, where you can specify the
+tag/branch used to download Airflow package from in github repository. You can
+also change the repository itself by adding --dockerhub-user and --dockerhub-repo flag values.
+
+Manually building the images
+----------------------------
+
+You can build the default production image with standard ``docker build`` command but they will only build
+default versions of the image and will not use the dockerhub versions of images as cache.
+
+
+CI images
+.........
+
+The following arguments can be used for CI images:
+
+* ARG PYTHON_BASE_IMAGE="python:3.6-slim-buster" - Base python image
+* ARG AIRFLOW_VERSION="2.0.0.dev0" - version of Airflow
+* ARG PYTHON_MAJOR_MINOR_VERSION="3.6" - major/minor version of Python (should match base image)
+* ARG DEPENDENCIES_EPOCH_NUMBER="2" - increasing this number will reinstall all apt dependencies
+* ARG KUBECTL_VERSION="v1.15.3" - version of kubectl installed
+* ARG KIND_VERSION="v0.6.1" - version of kind installed
+* ARG PIP_NO_CACHE_DIR="true" - if true, then no pip cache will be stored
+* ARG PIP_VERSION="19.0.2" - version of PIP to use
+* ARG HOME=/root - Home directory of the root user (CI image has root user as default)
+* ARG AIRFLOW_HOME=/root/airflow - Airflow's HOME (that's where logs and sqlite databases are stored)
+* ARG AIRFLOW_SOURCES=/opt/airflow - Mounted sources of Airflow
+* ARG PIP_DEPENDENCIES_EPOCH_NUMBER="3" - increasing that number will reinstall all PIP dependencies
+* ARG CASS_DRIVER_NO_CYTHON="1" - if set to 1 no CYTHON compilation is done for cassandra driver (much faster)
+* ARG AIRFLOW_CONTAINER_CI_OPTIMISED_BUILD="true" if set then PIP dependencies are installed from repo first
+      before they are reinstalled from local sources. This allows for incremental faster builds when
+      requirements change
```

Comment:
![image](https://user-images.githubusercontent.com/8811558/78026294-48423600-7353-11ea-9d09-788c86e8a000.png)

This looks weird. Maybe use backticks to wrap the args

Reply:
I will convert to tables. I was just lazy  :)

Reply:
Here is table. Hope it helps
```
Environment variable | Default value | Description
--|--|--
``PYTHON_BASE_IMAGE`` | ``python:3.6-slim-buster`` | Base python image
``AIRFLOW_VERSION`` | ``2.0.0.dev0`` | version of Airflow
``PYTHON_MAJOR_MINOR_VERSION`` | ``3.6`` | major/minor version of Python (should match base image)
``DEPENDENCIES_EPOCH_NUMBER`` | ``2`` | increasing this number will reinstall all apt dependencies
``KUBECTL_VERSION`` | ``v1.15.3`` | version of kubectl installed
``KIND_VERSION`` | ``v0.6.1`` | version of kind installed
``PIP_NO_CACHE_DIR`` | ``true`` | if true, then no pip cache will be stored
``PIP_VERSION`` | ``19.0.2`` | version of PIP to use
``HOME`` | ``/root`` | Home directory of the root user (CI image has root user as default)
``AIRFLOW_HOME`` | ``/root/airflow`` | Airflow's HOME (that's where logs and sqlite databases are stored)
``AIRFLOW_SOURCES`` | ``/opt/airflow`` | Mounted sources of Airflow
``PIP_DEPENDENCIES_EPOCH_NUMBER`` | ``3`` | increasing that number will reinstall all PIP dependencies
``CASS_DRIVER_NO_CYTHON`` | ``1`` | if set to 1 no CYTHON compilation is done for cassandra driver (much faster)
``AIRFLOW_CONTAINER_CI_OPTIMISED_BUILD`` | ``true`` if set then PIP dependencies are installed from repo first before they are reinstalled from local sources. This allows for incremental faster builds when requirements change
``AIRFLOW_REPO`` | ``apache/airflow`` | the repository from which PIP dependencies are installed (CI optimised)
``AIRFLOW_BRANCH`` | ``master`` | the branch from which PIP dependencies are installed (CI optimised)
``AIRFLOW_CI_BUILD_EPOCH`` | ``1`` | increasing this value will reinstall PIP dependencies from the repository from scratch
``AIRFLOW_EXTRAS`` | ``all`` | extras to install
``ADDITIONAL_PYTHON_DEPS`` | ```` | additional python dependencies to install
```

Reply:
Oh! Fantastic! Thanks! I was just about to convert them :)

Reply:
It's markdown not RST but I can convert them easily :)

Reply:
Sorry. I can update the comment in a bit.

Reply:
```
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| Environment variable                     | Default value              | Description                                                                            |
+==========================================+============================+========================================================================================+
| ``PYTHON_BASE_IMAGE``                    | ``python:3.6-slim-buster`` | Base python image                                                                      |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_VERSION``                      | ``2.0.0.dev0``             | version of Airflow                                                                     |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``PYTHON_MAJOR_MINOR_VERSION``           | ``3.6``                    | major/minor version of Python (should match base image)                                |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``DEPENDENCIES_EPOCH_NUMBER``            | ``2``                      | increasing this number will reinstall all apt dependencies                             |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``KUBECTL_VERSION``                      | ``v1.15.3``                | version of kubectl installed                                                           |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``KIND_VERSION``                         | ``v0.6.1``                 | version of kind installed                                                              |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``PIP_NO_CACHE_DIR``                     | ``true``                   | if true, then no pip cache will be stored                                              |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``PIP_VERSION``                          | ``19.0.2``                 | version of PIP to use                                                                  |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``HOME``                                 | ``/root``                  | Home directory of the root user (CI image has root user as default)                    |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_HOME``                         | ``/root/airflow``          | Airflow's HOME (that's where logs and sqlite databases are stored)                     |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_SOURCES``                      | ``/opt/airflow``           | Mounted sources of Airflow                                                             |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``PIP_DEPENDENCIES_EPOCH_NUMBER``        | ``3``                      | increasing that number will reinstall all PIP dependencies                             |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``CASS_DRIVER_NO_CYTHON``                | ``1``                      | if set to 1 no CYTHON compilation is done for cassandra driver (much faster)           |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_CONTAINER_CI_OPTIMISED_BUILD`` | ``true``                   | if set then PIP dependencies are installed from repo first before they are reinstalled |
|                                          |                            | from local sources. This allows for incremental faster builds when requirements change |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_REPO``                         | ``apache/airflow``         | the repository from which PIP dependencies are installed (CI optimised)                |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_BRANCH``                       | ``master``                 | the branch from which PIP dependencies are installed (CI optimised)                    |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_CI_BUILD_EPOCH``               | ``1``                      | increasing this value will reinstall PIP dependencies from the repository from scratch |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``AIRFLOW_EXTRAS``                       | ``all``                    | extras to install                                                                      |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
| ``ADDITIONAL_PYTHON_DEPS``               |                            | additional python dependencies to install                                              |
+------------------------------------------+----------------------------+----------------------------------------------------------------------------------------+
```